Library/Homebrew/dev-cmd/pr-pull.rb

```diff
@@ -105,6 +119,149 @@ def signoff!(pr, tap:, args:)
     end
   end
 
+  def determine_bump_subject(file, original_commit, path: ".", reason: nil)
+    full_path = Pathname.new(path)/file
+    formula_name = File.basename(file.chomp(".rb"))
+    old_file = Utils.popen_read "git", "-C", path, "show", "#{original_commit}:#{file}"
+
+    old_formula = begin
+                    Formulary.from_contents(formula_name, full_path, old_file, :stable)
+                  rescue FormulaUnavailableError
+                    nil
+                  end
+    new_formula = begin
+                    Formulary::FormulaLoader.new(formula_name, full_path).get_formula(:stable)
+                  rescue FormulaUnavailableError
+                    nil
+                  end
+
+    return "#{formula_name} #{new_formula.stable.version} (new formula)" unless old_formula
+    return "#{formula_name}: delete #{reason}" unless new_formula
+    return "#{formula_name} #{new_formula.stable.version}" if old_formula.stable.version != new_formula.stable.version
+    return "#{formula_name}: revision #{reason}" if old_formula.revision != new_formula.revision
+
+    "#{formula_name}: #{reason || "rebuild"}"
```

Comment:
might be nice to split the `reason` or `rebuild` case into an early return like the above.